src/object/srv_obj.c

```diff
@@ -177,43 +197,28 @@ obj_rw_reply(crt_rpc_t *rpc, int status, uint64_t epoch,
 	} else {
 		D_WARN("lost reply rpc %p\n", rpc);
 	}
+	if (!obj_rpc_is_fetch(rpc))
+		return;
 
-	if (obj_rpc_is_fetch(rpc)) {
-		if (orwo->orw_iod_sizes.ca_arrays != NULL) {
-			D_FREE(orwo->orw_iod_sizes.ca_arrays);
-			orwo->orw_iod_sizes.ca_count = 0;
-		}
-
-		if (orwo->orw_nrs.ca_arrays != NULL) {
-			D_FREE(orwo->orw_nrs.ca_arrays);
-			orwo->orw_nrs.ca_count = 0;
-		}
-
-		if (orwo->orw_iod_csums.ca_arrays != NULL) {
+	if (orwo->orw_iod_sizes.ca_arrays != NULL &&
+	    orwo->orw_iod_sizes.ca_arrays != (void *)&ioc->ioc_sizes[0]) {
+		D_FREE(orwo->orw_iod_sizes.ca_arrays);
+		orwo->orw_iod_sizes.ca_count = 0;
+	}
+ 
+	if (orwo->orw_nrs.ca_arrays != NULL &&
+	    orwo->orw_nrs.ca_arrays != (void *)&ioc->ioc_nrs[0]) {
+		D_FREE(orwo->orw_nrs.ca_arrays);
+		orwo->orw_nrs.ca_count = 0;
+	}
+	if (orwo->orw_data_sizes.ca_arrays != NULL &&
+	    orwo->orw_data_sizes.ca_arrays != (void *)&ioc->ioc_data_sizes[0]) {
+		D_FREE(orwo->orw_data_sizes.ca_arrays);
+		orwo->orw_data_sizes.ca_count = 0;
+ 	}
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs